src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs

```diff
@@ -166,6 +188,32 @@ protected override void Dispose(bool finalize)
             base.Dispose(finalize);
         }
 
+        private static List<Exception> Flatten(ICollection<Exception> exceptions)
+        {
+            var result = new List<Exception>();
+            foreach (var exception in exceptions)
+            {
+                AddException(exception, result);
+            }
+
+            return result;
+        }
+
+        private static void AddException(Exception exception, List<Exception> result)
+        {
+            if (exception is AggregateException aggregate)
```

Comment:
AggregateException has a built in `Flatten` method: https://source.dot.net/#System.Private.CoreLib/src/System/AggregateException.cs,389